src/domain/filtering/filter-link.ts

```diff
@@ -14,14 +14,14 @@ export const filterLink = (link: Link, filters: Filters): boolean => {
 
   if (link.isDNSRequest && filters.skipKubeDns) return false;
 
-  let ok = !filters.filters?.length;
-  filters.filters?.forEach((ff: FilterEntry) => {
-    const passed = filterLinkByEntry(link, ff);
-
-    ok = ok || passed;
-  });
-
-  return ok;
+  const filtered = filters.filters?.some(
+    ff => ff.negative && !filterLinkByEntry(link, ff),
+  );
+  return filtered
+    ? false
+    : !!filters.filters?.some(
+        ff => !ff.negative && filterLinkByEntry(link, ff),
+      );
```

Comment:
Could this be somehow optimized not to repeat itereations through `filters.filters`? Maybe smth like this:
```js
for (const ff of filters.filters) {
  const ffResult = filterLinkByEntry(link, ff);

  if (ff.negative && !ffResult) return false;
  if (!ff.negative && ffResult) return true;
}
```